src/main/java/com/cognifide/aemrules/checks/ContentResourceShouldBeNullCheckedCheck.java

```diff
@@ -0,0 +1,266 @@
+)
+public class ContentResourceShouldBeNullCheckedCheck extends BaseTreeVisitor implements
+    JavaFileScanner {
+
+    public static final String RULE_KEY = "AEM-18";
+
+    public static final String RULE_MESSAGE = "Always null check the returned value of Page.getContentResource() method";
+
+    public static final String METHOD_NAME = "getContentResource";
+
+    public static final String NOT_EQUAL = "NOT_EQUAL_TO";
+
+    public static final String EQUAL = "EQUAL_TO";
+
+    public static final String NON_NULL_METHOD = "nonNull";
+
+    public static final String ALL_NON_NULL_METHOD = "allNotNull";
+
+    public static final String IS_NULL_METHOD = "isNull";
+
+    private boolean returnOccurredInsideEqualNullCheck = false;
+
+    private boolean insideEqualNullCheck = false;
+
+    private boolean insideIfStatement = false;
+
+    private JavaFileScannerContext context;
+
+    private Map<String, Boolean> contentResources = new HashMap();
+
+    @Override
+    public void scanFile(JavaFileScannerContext context) {
+        this.context = context;
+        scan(context.getTree());
+    }
+
+    @Override
+    public void visitIfStatement(IfStatementTree tree) {
+        insideIfStatement = true;
+
+        if (isThisAResourceNullCheck(tree, NOT_EQUAL)) {
+            updateNullCheckedResources(tree, true);
+        } else if (isThisAResourceNullCheck(tree, EQUAL)) {
+            insideEqualNullCheck = true;
+        }
+        super.visitIfStatement(tree);
+        if (returnOccurredInsideEqualNullCheck) {
+            updateNullCheckedResources(tree, true);
+        }
+        cleanFlagsAfterIfStatement(tree);
+    }
+
+    @Override
+    public void visitAssignmentExpression(AssignmentExpressionTree tree) {
+        if (isGetContentResourceUsedOnResource(tree)) {
+            contentResources.put(tree.variable().firstToken().text(), false);
+        }
+        super.visitAssignmentExpression(tree);
+    }
+
+    @Override
+    public void visitMethodInvocation(MethodInvocationTree tree) {
+        externalLibraryNullChecks(tree);
+        equalsNullCheck(tree);
+        chainedMethodsNullCheck(tree);
+        super.visitMethodInvocation(tree);
+    }
+
+    @Override
+    public void visitVariable(VariableTree tree) {
+        if (isResourceInitialized(tree)) {
+            shouldBeNullChecked(tree);
+        }
+        super.visitVariable(tree);
+    }
+
+    @Override
+    public void visitClass(ClassTree tree) {
+        contentResources.clear();
+        super.visitClass(tree);
+    }
+
+    @Override
+    public void visitMethod(MethodTree tree) {
+        super.visitMethod(tree);
+        returnOccurredInsideEqualNullCheck = false;
+    }
+
+    @Override
+    public void visitReturnStatement(ReturnStatementTree tree) {
+        if (insideEqualNullCheck) {
+            returnOccurredInsideEqualNullCheck = true;
+        }
+        super.visitReturnStatement(tree);
+    }
+
+    private void shouldBeNullChecked(VariableTree tree) {
```

Comment:
Name of the method indicates that should return boolean. In this case I would rather move this logic to line 117
isResourceInitialized(tree) && isGetContentResourceUsedOnPage(tree) {
contentResources.put(tree.simpleName().name(), false);
}

Reply:
done